addons/survey/models/survey.py

```diff
@@ -0,0 +1,1096 @@
+        string="Public link (html version)")
+    print_url = fields.Char(compute='_compute_print_url',
+        string="Print link")
+    result_url = fields.Char(compute='_compute_result_url',
+        string="Results link")
+    email_template_id = fields.Many2one('mail.template',
+        string='Email Template')
+    thank_you_message = fields.Html(translate=True,
+        help="This message will be displayed when survey is completed")
+    quizz_mode = fields.Boolean(string='Quiz mode')
+    active = fields.Boolean(default=True)
+
+    ## Function fields ##
+
+    @api.multi
+    def _compute_designed(self):
+        for survey in self:
+            if not survey.page_ids or not survey.page_ids.filtered(lambda page: page.question_ids.ids).mapped('question_ids'):
+                survey.designed = False
+            else:
+                survey.designed = True
+
+    @api.multi
+    def _compute_tot_sent_survey(self):
+        """ Returns the number of invitations sent for this survey, be they
+        (partially) completed or not """
+        survey_data = self.env['survey.user_input'].read_group([('survey_id', 'in', self.ids), ('input_type', '=', 'link')], ['survey_id'], ['survey_id'])
+        result = dict((data['survey_id'][0], data['survey_id_count']) for data in survey_data)
+        for survey in self:
+            survey.tot_sent_survey = result.get(survey.id, 0)
+
+    @api.multi
+    def _compute_tot_start_survey(self):
+        """ Returns the number of started instances of this survey, be they
+        completed or not """
+        survey_data = self.env['survey.user_input'].read_group(['&', ('survey_id', 'in', self.ids), '|', ('state', '=', 'skip'), ('state', '=', 'done')], ['survey_id'], ['survey_id'])
+        result = dict((data['survey_id'][0], data['survey_id_count']) for data in survey_data)
+        for survey in self:
+            survey.tot_start_survey = result.get(survey.id, 0)
+
+    @api.multi
+    def _compute_tot_comp_survey(self):
+        """ Returns the number of completed instances of this survey """
+        survey_data = self.env['survey.user_input'].read_group([('survey_id', 'in', self.ids), ('state', '=', 'done')], ['survey_id'], ['survey_id'])
+        result = dict((data['survey_id'][0], data['survey_id_count']) for data in survey_data)
+        for survey in self:
+            survey.tot_comp_survey = result.get(survey.id, 0)
+
+    @api.multi
+    def _compute_public_url(self):
+        """ Computes a public URL for the survey """
+        base_url = '/' if self.env.context.get('relative_url') else self.env['ir.config_parameter'].get_param('web.base.url')
+        for survey in self:
+            survey.public_url = urljoin(base_url, "survey/start/%s" % slug(survey))
+
+    @api.multi
+    def _compute_public_url_html(self):
+        """ Computes a public URL for the survey (html-embeddable version)"""
+        self._compute_public_url()
```

Comment:
Do we need explicit call to _compute_public_url, don't think survey.public_url will do the same thing